jupyter_drives/handlers.py

```diff
@@ -1,24 +1,91 @@
+"""
+Module with all of the individual handlers, which will return the results to the frontend.
+"""
 import json
+import logging 
+import traceback
+from typing import Optional
 
 from jupyter_server.base.handlers import APIHandler
 from jupyter_server.utils import url_path_join
 import tornado
+import traitlets
 
-class RouteHandler(APIHandler):
-    # The following decorator should be present on all verb methods (head, get, post,
-    # patch, put, delete, options) to ensure only authorized user can request the
-    # Jupyter server
+from .base import MANAGERS, DrivesConfig
+from .managers.manager import JupyterDrivesManager
+
+NAMESPACE = "jupyter-drives"
+
+class JupyterDrivesAPIHandler(APIHandler):
+    """
+    Base handler for jupyter-drives specific API handlers
+    """
+    def initialize(self, logger: logging.Logger, manager: JupyterDrivesManager):
+        self._jp_log = logger
+        self._manager = manager
+
+    def write_error(self, status_code, **kwargs):
+        """
+        Override Tornado's RequestHandler.write_error for customized error handlings
+        This method will be called when an exception is raised from a handler
+        """
+        self.set_header("Content-Type", "application/json")
+        reply = {"error": "Unhandled error"}
+        exc_info = kwargs.get("exc_info")
+        if exc_info:
+            e = exc_info[1]
+            if isinstance(e, tornado.web.HTTPError):
+                reply["error"] = e.reason
+                if hasattr(e, "error_code"):
+                    reply["error_code"] = e.error_code
+            else:
+                reply["error"] = "".join(traceback.format_exception(*exc_info))
+        self.finish(json.dumps(reply))
+
+class ListJupyterDrives(JupyterDrivesAPIHandler):
+    """
+    Returns list of available drives.
+    """
+    # Later on, filters can be added for the listing 
     @tornado.web.authenticated
-    def get(self):
-        self.finish(json.dumps({
-            "data": "This is /jupyter-drives/get-example endpoint!"
-        }))
+    async def get(self):
+        drives, error = await self._manager.list_drives()
+        self.finish(json.dumps(drives))
+        logging.debug(error)
```

Comment:
Let's use this extension logger to scope this message:

```suggestion
        get_logger().debug(error)
```